tests/test_emr/test_emr_boto3.py

```diff
@@ -476,6 +476,118 @@ def test_run_job_flow_with_instance_groups():
+            "Action": {
+                "SimpleScalingPolicyConfiguration": {
+                    "AdjustmentType": "CHANGE_IN_CAPACITY",
+                    "ScalingAdjustment": 1,
+                    "CoolDown": 300,
+                }
+            },
+            "Trigger": {
+                "CloudWatchAlarmDefinition": {
+                    "ComparisonOperator": "LESS_THAN",
+                    "EvaluationPeriods": 1,
+                    "MetricName": "YARNMemoryAvailablePercentage",
+                    "Namespace": "AWS/ElasticMapReduce",
+                    "Period": 300,
+                    "Threshold": 15.0,
+                    "Statistic": "AVERAGE",
+                    "Unit": "PERCENT",
+                    "Dimensions": [{"Key": "JobFlowId", "Value": "${emr.clusterId}"}],
+                }
+            },
+        }
+    ],
+}
+
+
+@mock_emr
+def test_run_job_flow_with_instance_groups_with_autoscaling():
+    input_groups = dict((g["Name"], g) for g in input_instance_groups)
+
+    input_groups["core"]["AutoScalingPolicy"] = auto_scaling_policy
+    input_groups["task-1"]["AutoScalingPolicy"] = auto_scaling_policy
+
+    client = boto3.client("emr", region_name="us-east-1")
+    args = deepcopy(run_job_flow_args)
+    args["Instances"] = {"InstanceGroups": input_instance_groups}
+    cluster_id = client.run_job_flow(**args)["JobFlowId"]
+    groups = client.list_instance_groups(ClusterId=cluster_id)["InstanceGroups"]
+    for x in groups:
+        y = deepcopy(input_groups[x["Name"]])
+        if "AutoScalingPolicy" in y:
+            x["AutoScalingPolicy"]["Status"]["State"].should.equal("ATTACHED")
+            returned_policy = deepcopy(x["AutoScalingPolicy"])
+            auto_scaling_policy_with_cluster_id = _patch_cluster_id_placeholder_in_autoscaling_policy(
+                y["AutoScalingPolicy"], cluster_id
+            )
+            del returned_policy["Status"]
+            returned_policy.should.equal(auto_scaling_policy_with_cluster_id)
+
+
+@mock_emr
+def test_put_remove_auto_scaling_policy():
+    input_groups = dict((g["Name"], g) for g in input_instance_groups)
+    client = boto3.client("emr", region_name="us-east-1")
+    args = deepcopy(run_job_flow_args)
+    args["Instances"] = {"InstanceGroups": input_instance_groups}
+    cluster_id = client.run_job_flow(**args)["JobFlowId"]
+
+    core_instance_group = [
+        ig
+        for ig in client.list_instance_groups(ClusterId=cluster_id)["InstanceGroups"]
+        if ig["InstanceGroupType"] == "CORE"
+    ][0]
+
+    resp = client.put_auto_scaling_policy(
+        ClusterId=cluster_id,
+        InstanceGroupId=core_instance_group["Id"],
+        AutoScalingPolicy=auto_scaling_policy,
+    )
+
+    auto_scaling_policy_with_cluster_id = _patch_cluster_id_placeholder_in_autoscaling_policy(
+        auto_scaling_policy, cluster_id
+    )
+    del resp["AutoScalingPolicy"]["Status"]
```

Comment:
Let's assert the `Status` arg has the right value, before discarding it

Reply:
Done.